git/operations.go

```diff
@@ -22,12 +22,21 @@ const trace = false
 // Env vars that are allowed to be inherited from the os
 var allowedEnvVars = []string{"http_proxy", "https_proxy", "no_proxy", "HOME"}
 
+type gitCmd struct {
```

Comment:
This approach (using a struct to package up arguments) works well in some situations:

 - providing optional arguments, i.e., where some of them may be omitted (and thereby get a zero value)
 - bundling several closely related arguments, all of which must be supplied, to declutter

Here it's not really either case; but, it _could_ be the first if the struct had only the things that were optional. Looking at `execGitCmd`, those are dir, env, and out. The args and context are always required, so I'd expect those to be their own arguments.